packages/xod-arduino/platform/program.tpl.cpp

```diff
@@ -250,27 +226,28 @@ void runTransaction() {
 
           {{#if patch.raisesErrors}}
 #if defined(XOD_DEBUG) || defined(XOD_SIMULATION)
-            uint8_t prevOwnError = node_{{ id }}.ownError;
+            ErrorFlags previousErrorFlags = node_{{ id }}.errorFlags;
 #endif
-            // give the node a chance to recover from it's own previous error
-            node_{{ id }}.ownError = 0;
+            // give the node a chance to recover from it's own previous errors
+            node_{{ id }}.errorFlags = 0;
           {{/if}}
 
             {{ ns patch }}::evaluate(&ctxObj);
 
           {{#if patch.raisesErrors}}
 #if defined(XOD_DEBUG) || defined(XOD_SIMULATION)
-            if (prevOwnError && !node_{{ id }}.ownError) {
-                // report that the node recovered from error
-                detail::printErrorToDebugSerial({{ id }}, 0);
+            if (previousErrorFlags != node_{{ id }}.errorFlags) {
```

Comment:
I think the `#if` guard can be omitted. The C++ optimizer should take care to throw

```cpp
if (doesNotMatterWhat) {
    emptyFunction();
}
```

Reply:
@evgenykochetkov